repo/blob/azure/azure_storage.go

```diff
@@ -0,0 +1,273 @@
+// Package azure implements Azure Blob Storage.
+package azure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/efarrer/iothrottler"
+	"github.com/pkg/errors"
+	"gocloud.dev/gcerrors"
+
+	"github.com/kopia/kopia/internal/retry"
+	"github.com/kopia/kopia/repo/blob"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	gblob "gocloud.dev/blob"
+	"gocloud.dev/blob/azureblob"
+)
+
+const (
+	azStorageType   = "azure"
+	writerChunkSize = 1 << 20
+)
+
+type azStorage struct {
+	Options
+
+	ctx context.Context
+
+	bucket *gblob.Bucket
+
+	downloadThrottler *iothrottler.IOThrottlerPool
+	uploadThrottler   *iothrottler.IOThrottlerPool
+}
+
+func (az *azStorage) GetBlob(ctx context.Context, b blob.ID, offset, length int64) ([]byte, error) {
+	attempt := func() (interface{}, error) {
+		// TODO: Support chunked download
```

Comment:
What's needed here is range reads. Is that what you mean?

Reply:
Yes. I think better to use NewRangeReader()